### PR TITLE
br: make download metadata concurrency adjustable

### LINKED ISSUE #45511: [Dynamic Regions] in restore point , br oomkilled
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1. br 4c16g
2. br restore point to partitioned-raft-kv cluster 

### 3. What did you see instead (Required)
1. br oomed 
[Sat Jul 22 01:07:21 2023] Memory cgroup out of memory: Killed process 2748779 (br-fixed-3) total-vm:18009712kB, anon-rss:16348072kB, file-rss:1148kB, shmem-rss:0kB, UID:0 pgtables:32468kB oom_score_adj:-997
### 4. What is your TiDB version? (Required)
master

### PR DESCRIPTION
**Pull request by Leavrth**
### What problem does this PR solve?

Issue Number: close #45511 

Problem Summary:
There are metadatas at most 10 MB each, and keep 512 metadatas in the memory at once. This will make GC too late.
### What is changed and how it works?
make the 512 adjustable by config
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/restore/log_client.go [hunk 1]
```diff
@@ -174,7 +175,7 @@ func (rc *logFileManager) createMetaIterOver(ctx context.Context, s storage.Exte
 		return meta, nil
 	}
 	reader := iter.Transform(namesIter, readMeta,
-		iter.WithChunkSize(readMetaBatchSize), iter.WithConcurrency(readMetaConcurrency))
+		iter.WithChunkSize(rc.metadataDownloadBatchSize), iter.WithConcurrency(rc.metadataDownloadBatchSize))
```
Comment (3pointer) [inline br/pkg/restore/log_client.go:178]: maybe we need a general name here

File: br/pkg/task/common.go [hunk 2]
```diff
@@ -92,6 +92,9 @@ const (
 	flagCipherKey     = "crypter.key"
 	flagCipherKeyFile = "crypter.key-file"
 
+	flagMetadataDownloadBatchSize    = "metadata-download-batch-size"
+	defaultMetadataDownloadBatchSize = 64
```
Comment (3pointer) [inline br/pkg/task/common.go:96]: will it makes restore slow by default? maybe we can set it to 128 to keep the compatibility

**Review by YuJuncen — APPROVED** [review]

**Review by 3pointer — APPROVED** [review]

**Pull request merged**